From 9d998db84c8b2b5e18d5a6d00e3f7fee43d726c7 Mon Sep 17 00:00:00 2001
From: Richard Muzik <richard.muzik@turris.com>
Date: Wed, 6 May 2026 14:37:39 +0200
Subject: [PATCH] rust: Change handling RUSTFLAGS

This is taken from [1] and helped compiling peacomp for OpenWrt 25.12.

[1] https://github.com/openwrt/packages/issues/24375
---
 lang/rust/rust-values.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lang/rust/rust-values.mk b/lang/rust/rust-values.mk
index e1dbbbcca..bf7c448be 100644
--- a/lang/rust/rust-values.mk
+++ b/lang/rust/rust-values.mk
@@ -99,7 +99,7 @@ CARGO_PKG_CONFIG_VARS= \
 	CARGO_PROFILE_RELEASE_PANIC=unwind \
 	CARGO_PROFILE_RELEASE_RPATH=false \
 	CARGO_TARGET_$(subst -,_,$(call toupper,$(RUSTC_TARGET_ARCH)))_LINKER=$(TARGET_CC_NOCACHE) \
-	RUSTFLAGS="$(CARGO_RUSTFLAGS)" \
+	CARGO_TARGET_$(subst -,_,$(call toupper,$(RUSTC_TARGET_ARCH)))_RUSTFLAGS="$(CARGO_RUSTFLAGS)" \
 	TARGET_CC=$(TARGET_CC_NOCACHE) \
 	TARGET_CFLAGS="$(TARGET_CFLAGS) $(RUSTC_CFLAGS)"
 
-- 
2.53.0

